openmc/data/function.py

```diff
@@ -432,6 +488,63 @@ def functions(self, functions):
         self._functions = functions
 
 
+class Piecewise(EqualityMixin):
+    """Piecewise composition of multiple functions.
+
+    This class allows you to create a callable object which is composed
+    of multiple other callable objects, each applying to a specific interval
+
+    Parameters
+    ----------
+    functions : Iterable of Callable
+        Functions which are to be combined in a piecewise fashion
+    breakpoints : Iterable of float
+        The breakpoints between each function. The functions
+        in the functions attribute must be provided in order of
+        increasing intervals.
```

Comment:
Maybe change to something like: "The values of the dependent variable that define the domain of each function. The *i*th and *(i+1)*th values are the limits of the domain of the *i*th function. Values must be monotonically increasing."